### ISSUE TITLE
[await-async-utils] false positive promise all

### ISSUE THREAD

**Issue by Mikeysauce** [testing-library/eslint-plugin-testing-library#227]
Hello,

Take the following example:

```
await Promise.all([
    waitFor(() =>
      expect(getByText('first-item')).toBeTruthy()
    ),
    waitFor(() =>
      expect(getByText('second-item')).toBeTruthy()
    ),
  ]);
```

This is flagged up as an error: "Promise returned from `waitFor` must be handled", which I believe is a false positive, as the Promise is indeed being handled by the Promise.all.

Thanks

**Comment by gndelia** [comment]
Hi Mikey! 
If you are waiting for of those two to appear, perhaps you could consider `findByText` as an alternative

```ts
await Promise.all([findByText('first-item'), findByText('second-item')])
```

As of the code itself, yeah it sounds like a bug

Thanks for opening the ticket!

**Comment by Mikeysauce** [comment]
Thanks gndelia,

I'm in agreement with you regarding findBy being more suitable here with it having a built in waitFor behaviour, but some colleagues find the intent of waitFor to be more explicit and so here we are unfortunately. I appreciate the quick reply!
